### ISSUE TITLE
A closed Recurring Task duplicates when reopened

### ISSUE THREAD

**Issue by supertoothy** [getting-things-gnome/gtg#636]
1. Mark a recurring task as done
2. Go to CLOSED tasks and Mark as not done
3. Go to OPEN tasks - there is a duplicate, of course. The old task, alongside the new
4. Mark the old task as done
5. Now there are two new recurring tasks in the list

This happened when I accidentally marked a daily recurring task as **_done_** twice in the day. Then I went to CLOSED to see if marking it as _**not done**_ will replace the original.

**Comment by zeddo123** [comment]
Yes, I have known this to happen. Basically, because I wanted to keep the history of tasks, whenever a repeating task is mark as done it gets duplicated. The first goes to Done and the new one has the next date.

Needless to say, we have to look into it and see how we might fix it.

**Comment by nekohayo** [comment]
I guess it's either checking that there still is a "parent" (so to speak) recurring task open, opening that as a task editor window so the user can see it, while maybe deleting the closed task in the background or marking it non-recurring. But I guess this might require the ability to easily re-schedule the recurrence.

Or maybe doing that and showing a choice dialog to the user explaining that the closed task had X  due date, would you like to update the remaining recurring task to match that old due date instead of the future due date...?

Or another approach could be to simply remove the recurrence of the closed instances of a task, so users can reopen them normally but they will still end up with the duplicate.

**Comment by zeddo123** [comment]
I was thinking maybe we could "link" the new task and the older one together by having the ID of the new task in the old task (along side the recurring information). So that when the old task is marked as not done, we can figure out which task we should delete.

(sorry for not being active lately, my studies caught up with me.)

**Comment by diegogangl** [comment]
> (sorry for not being active lately, my studies caught up with me.)
@zeddo123  no problem. Stress caught up with me and I had to cut back on gtg and several other things for the past 2 months -_-

The link solution sounds good. Don't be afraid to think of more "invasive" ideas if you want too, we can try them in the new core branch (there's nothing about recurring implemented there)

**Comment by pentux-GitHub** [comment]
Housekeeping: the recurrence RFC #645 covers this explicitly in its "History and duplicate tasks" section, which references this issue. It seems safe to close in favor of tracking it there.
